documentation/binding-data/tutorial-flow-data-provider.asciidoc

```diff
@@ -1,50 +1,109 @@
 ---
-title: Showing a List of Data with Data Providers
+title: Binding Data Items to Components
 order: 5
 layout: page
 ---
 
-= Showing a List of Data with Data Providers
+= Binding Data Items to Components
 
-Many applications present the user with a list of items from which they can select one or more items to work on. Example lists include inventory records to survey, messages requiring a response, or blog drafts to edit or publish.
+You often have lists of items that you want to display, and possibly let the user select one or more of them.
+You can use basic components such as HTML elements to display such lists,
+or you can use components especially designed for the purpose, such as
+`Grid`, `ComboBox`, and `ListBox`.
 
-A listing component is a component that:
+[source, java]
+----
+// Create a listing component for a bean type
+Grid grid = new Grid(Person.class);
 
-* Displays one or several properties from a list of items, 
-* Allows the user to inspect the data and mark items as selected, and 
-* Optionally, allows the user to edit items directly in the component.
+// Sets items using vararg beans
+grid.setItems(
+        new Person("George Washington", 1732),
+        new Person("John Adams", 1735),
+        new Person("Thomas Jefferson", 1743),
+        new Person("James Madison", 1751)
+);
+----
 
-There are a number of officially-supported listing components, such as `ListBox` and `ComboBox`. Each component has its own API to configure exactly how the data is represented and manipulated. 
+All listing components in Vaadin have a number of `setItems()` methods to define the items to display.
+Items can be basic objects, such as strings or numbers, or POJOs, such as DTOs or JPA entities, from your domain model.
+The easiest way is to provide a `java.util.List` of objects to be shown in such a component.
 
-All list components have a `setItems` method to define which items display. They also have the `DataProvider` interface for more fine-grained control of the displayed data.
+If the number of items is large and reserves a lot of memory, `Grid` and `ComboBox` allow you to do a lazy data binding using callbacks to fetch only the required set of items from the back-end.
```

Comment:
Yes, this is a current situation, both are providing the lazy loading.

Reply:
The main question here was : how it will be updated in the future when something will be changed ?
Isn't it better to put component specific docs in the component specific sections ?

Reply:
I checked a `components` section and particular component's pages (demo examples), but IMO they do not fit for this binding API docs. In fact, components demos do not include any common description, they just contain a code examples, how to do this, or how to do that. `component` section keeps an information about components features itself, not about any binding API, and I really don't want to create a new section there just to describe a particular usage case for a certain component.